etc/picongpu/crusher-ornl/batch_pipe.tpl

```diff
@@ -0,0 +1,247 @@
+# Otherwise, the current distribution strategy in openpmd-pipe is to simply
+# subdivide each dataset into n equal-sized hyperslabs.
+# As a consequence, communication will not necessarily happen within a node.
+# If however using that branch, the following environment variable advises the
+# openpmd-pipe script to load only data chunks originating from the same
+# host(name), using a bin-packing approach within hosts (irrelevant here since
+# there is only one reader per host), and failing with a runtime error if not
+# all chunks could be distributed this way (either because of bad scheduling or
+# because the writer did not send hostname information). Instead of "fail", a
+# secondary fallback strategy could be specified here
+# (e.g. hostname_binpacking_binpacking).
+
+# Implicit node-level aggregation via streaming is independent from distribution
+# strategy that is used, since the deciding factor is that there is only one
+# instance of openpmd-pipe writing data per node. It does not matter that this
+# data does not necessarily stem from the same node.
+export OPENPMD_CHUNK_DISTRIBUTION=hostname_binpacking_fail
+
+export MPICH_OFI_NIC_POLICY=NUMA # The default
+
+if [ $node_check_err -eq 0 ] || [ $run_cuda_memtest -eq 0 ] ; then
+    # Run PIConGPU
+    echo "Start PIConGPU."
+    test $n_broken_nodes -ne 0 && exclude_nodes="-x./bad_nodes.txt"
+
+    echo '!TBG_inconfig_pipe' | tee inconfig.json
+    echo '!TBG_outconfig_pipe' | tee outconfig.json
+
+    mkdir -p openPMD
+
+    export MPICH_OFI_CXI_PID_BASE=0
+    # Corresponds to the cores 0, 8, 16, 24, 32, 40, 48, 56
+    # (first core in each L3 cache group)
+    mask="0x101010101010101"
+
+    srun                                          \
+      --ntasks !TBG_nodes                         \
+      --nodes !TBG_nodes                          \
+      $exclude_nodes                              \
+      --ntasks-per-node=1                         \
+      --gpus-per-node=0                           \
+      --cpus-per-task=!TBG_coresPerPipeInstance   \
+      --cpu-bind=verbose,mask_cpu:$mask           \
+      openpmd-pipe                                \
+        --infile "!TBG_streamdir"                 \
+        --outfile "!TBG_dumpdir"                  \
+        --inconfig @inconfig.json                 \
+        --outconfig @outconfig.json               \
+        > ../pipe.out 2> ../pipe.err              &
+
+    sleep 1
+
+    # Need threaded MPI for SST on ECP systems
+    export PIC_USE_THREADED_MPI=MPI_THREAD_MULTIPLE
+    # Enable workaround as described here:
+    # https://github.com/ornladios/ADIOS2/blob/master/docs/user_guide/source/advanced/ecp_hardware.rst
+    export PIC_WORKAROUND_CRAY_MPI_FINALIZE=1
+    export MPICH_OFI_CXI_PID_BASE=$((MPICH_OFI_CXI_PID_BASE+1))
+    # Corresponds to cores 1-8 (exclude first core) in each L3 cache group
+    mask=0xfe,0xfe00,0xfe0000,0xfe000000,0xfe00000000,0xfe0000000000,0xfe000000000000,0xfe00000000000000
```

Comment:
is this really required, this should be done by the batch system, or is this required because we use two times srun.

Reply:
This is required because of the two sruns. If you don't specify this, the job will hang up.